src/ape/contracts/base.py

```diff
@@ -842,6 +843,35 @@ def at(self, address: AddressType, txn_hash: Optional[str] = None) -> ContractIn
             address, self.contract_type, txn_hash=txn_hash
         )
 
+    @cached_property
+    def source_path(self) -> Optional[Path]:
+        """
+        Returns the path to the local contract if determined that this container
+        belongs to the active project.
+        """
+        contract_name = self.contract_type.name
+        source_id = self.contract_type.source_id
+        if not (contract_name and source_id):
+            return None
+
+        contract_container = self.project_manager._get_contract(contract_name)
+        if not (
+            contract_container
+            and contract_container.contract_type.runtime_bytecode
+            and self.contract_type.runtime_bytecode
+        ):
+            return None
+
+        contract_bytecode = contract_container.contract_type.runtime_bytecode.bytecode
+        self_bytecode = self.contract_type.runtime_bytecode.bytecode
+        if not (contract_bytecode and self_bytecode):
+            return None
+
+        if contract_bytecode == self_bytecode:
```

Comment:
This is not always a sufficient condition to check the origin matches the compiled code

Reply:
Can we use the ABI to verify equality?

Reply:
I think ABI may not work when using local proxies

Reply:
that's decently solid, we definitely know a lot more about locally-tracked deployments than we do about on-chain ones, and the majority of the time (probably all the time?) we are going to want the source id for those types of contracts

perhaps something a bit more interesting to be said of contracts sourced from dependency manifests

Reply:
@sabotagebeats pointed out that this isn't good because if I want to load the contract into `boa`, it may not be deployed.

So, thinking we should compare `source_id`s and call it a day